### DISCUSSION TITLE
HyperHDR hardware diagram, will this enable 4K120 / VRR / DV?

### DISCUSSION THREAD

**Discussion by Origin-of-Symmetry** [awawa-dev/HyperHDR#989] [General]
Let me start with a big thumbs up for @awawa-dev and others contributing to HyperHDR 👍  I have read many posts here on github to get an understanding of the parts and configuration needed for my own HyperHDR setup, thank you!

I still need to purchase a HDMI splitter with scaler. The 4K60 HDMI 2.0 Ezcoo EZ-SP12H2 is mentioned a lot. However I am tempted to go for the [EZ-SP12H21](https://www.amazon.de/-/en/Splitter-HDCP2-3-Vision-Scaler-Monitors/dp/B0C739QQ4Q?crid=95RWKFVW55GB&dib=eyJ2IjoiMSJ9.ttibecTQ70kNCZdUUBpKLiVZRgrM_2rTwhKeXU9fenxNfRAoih85yzM3m2ypl9X71MvHut1xA706aTGSYjpX4H-6ldmCSeh9eK35vlpFMN1aLDcSv6neYUAZWGhiSh-FTS78OJ3L50DNSxr4Ksk4iOErfuMaWZbgZpS6WRqhL6Y2CEBSrwgffAWOXEqr6DIXfRKyMnKBZYiK6rM8qNlzDz4Fpl64Mv5Ybr1I-t0Y3ss.vi_8ba68k8c7Wk9ZUmR7JbaiCB3Me320IBqs4XiGSWQ&dib_tag=se&keywords=ezcoo+splitter&nsdOptOutParam=true&qid=1731926200&sprefix=ezcoo+splitte%2Caps%2C90&sr=8-4) with HDMI 2.1 4K120 output. 

Am I correct that my Xbox Series X is able to output 4K120 with VRR and Dolby Vision by using the EZ-SP12H21 with the MS2130 grabber (with [F3 experimental firmware](https://github.com/awawa-dev/HyperHDR/discussions/729#discussioncomment-8126044) which should unlock 1080P120 input from the Ezcoo to MS2130)?

I have created a HyperHDR hardware diagram for my setup. Are there any flaws in my design?

![HyperHDR hardware diagram](https://github.com/user-attachments/assets/ead6ea6b-d8c4-4ecf-84d6-cd9ad7b378f7)

**Comment by awawa-dev** [comment]
Hi @Origin-of-Symmetry 
A very precise diagram 👍  If HyperHDR is the only application running on the RPi4, you can even choose the RPi4 1GB version, although this is probably a less promising solution for the future.

The ms2130 accepts incoming (not recording) a 1080p120 signal but only for certain color formats: you have to find it in the original ms2130 thread. But it definitely won't accept, for example, RGB444 because it simply won't have enough bandwidth. It also won't support VRR if the console uses it. Then there's one of the grabbers listed here: https://github.com/awawa-dev/HyperHDR/discussions/967, the cheapest of which is Hagibis 4K THB05 (pay attention to my remarks in that thread: https://github.com/awawa-dev/HyperHDR/discussions/967#discussioncomment-11124448 because it has some flaws).

As for the Ezcoo version for 4k120Hz, I haven't tested it, although on the other hand, none of the users who probably had it in their sights have complained about it here. 

Since you have an LG, there is theoretically an alternative solution in the form of Piccap, although if your TV has a newer firmware you have to wait for a new version of root. This allows you to get rid of most of the hardware, and the latest developments that you can follow in this thread: https://github.com/awawa-dev/HyperHDR/pull/896 have made HyperHDR now offer very good color quality for it that you will not find in any other software.

**Reply by Origin-of-Symmetry** [reply]
Thank you @awawa-dev 

Piccap seems like a great option, however I just recently purchased the TV. Unfortunately it has a recent firmware version that can't be rooted at this time.

Since you mentioned that the ms2130 won't accept RGB444, I've checked the Xbox HDMI input signal as reported on screen by my LG TV. It looks like my Xbox is not outputting 4:4:4:

DV 4k/120: YCBCR422 8b 4L8 Dolby Vision
DV 4k/60: YCBCR422 8b TM Dolby Vision

HDR 10 4k/120: RGB 10b 4L10 HDR10
HDR 10 4k/60: YCBCR420 10b TM HDR10

4k/120 SDR: RGB 10b 4L10 SDR
4k/60 SDR: YCBCR420 10b TM SDR

Although I'm not sure what bandwidth RGB 10b 4L10 uses, all other signals seem to be 4:2:0 or 4:2:2.

1080p/120 (after Ezcoo resolution scaling), 8 bit, 4:2:2 adds up to a total bandwidth of 5.94 Gbps according to [this calculator](https://www.murideo.com/hdmi-21-bandwidth-calculator.html), or 8.91 Gbps at 12 bit. So it stays within the HDMI 1.4 bandwidth limit of 10.2 Gbps.

![Bandwidth](https://github.com/user-attachments/assets/3c7c90bd-0123-4015-9c49-c0c1997a8a71)

Could I therefore make the assumption that the ms2130 grabber can accept the (Ezcoo down scaled) input signals with HDR/DV enabled at 120Hz? Or would it still be best to look out for an alternative video grabber?

**Reply by awawa-dev** [reply]
As you can see here: https://github.com/awawa-dev/HyperHDR/discussions/499#discussioncomment-4961638 it can accept even 12bit YCBCR420, at least it worked in my simply setup with gtx1060. You can give it a try but hard to predict a result in your case. Ezcoo will downscale 4k to 1080p to reduce the bandwidth if you enable the scaler.

**Reply by DavidinCT** [reply]
@awawa-dev This diagram @Origin-of-Symmetry created was so great this cleared up all my questions, the spotted things in my head on this project.  It would have cleared up all the 50 questions I asked you. :)

This is the exact setup I am going to do besides using different inputs (output from all my consoles and my AVR) and a different video grabber but, everything else is exactly what I plan on setting up.

You should use this diagram for examples it might clear up any questions on a basic setup.

The images you sent to gave me more questions and this image cleared it all up.
